src/GraphQLSchemaLinter.php

```diff
@@ -79,7 +79,10 @@ public function getVersion() {
     if (!empty($this->dependencyVersions)) {
 
       foreach ($this->dependencyVersions as $name => $required) {
-        list($err, $dep_version, $stderr) = exec_manual('cd .arcanist 2>&1 >/dev/null && npm list --depth=0 2>/dev/null | grep "%C" | cut -d "@" -f 3 | tr -d "[:space:]"', $name);
+        $binary_path = $this->getDefaultBinary();
+        $pos = strrpos($binary_path, '/');
+        $binary_path_parent = $pos === false ? $url : substr($binary_path, 0, $pos + 1);
+        list($err, $dep_version, $stderr) = exec_manual('cd %s 2>&1 >/dev/null && npm list --depth=0 2>/dev/null | grep "%C" | cut -d "@" -f 3 | tr -d "[:space:]"', $binary_path_parent, $name);
```

Comment:
I did a little more reading, and I think we could improve this so that it doesn't involve the `cd` step.

```php
list($stdout) = newv('ExecFuture', $args)
    ->setCWD($bin_root)
    ->resolvex();
```

Further, I think you could clean up the command line itself by requesting JSON output and parsing that in the PHP code (by passing `--json` to `npm list`). There are good examples of JSON output parsing in our linters.